### PR TITLE
Add io.github.vikdevelop.SaveDesktop

### PR DESCRIPTION
**Pull request by vikdevelop**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.github.vikdevelop.SaveDesktop.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    # for logout from session (Xfce)
+    - --talk-name=org.xfce.SessionManager
+modules:
+  - name: savedesktop
+    buildsystem: simple
+    build-commands:
+       - chmod +x src/main_window.py
+       - chmod +x run.sh
+       - install -D -t /app/bin run.sh
+       - install -D -t /app src/main_window.py
```
Comment (hfiguiere) [inline com.github.vikdevelop.SaveDesktop.yaml:24]:
```suggestion
       - install -Dm755 -t /app/bin run.sh
       - install -Dm755 -t /app src/main_window.py
```

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.SaveDesktop

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.SaveDesktop

**Comment by vikdevelop** [comment]
bot, build io.github.vikdevelop.SaveDesktop

File: io.github.vikdevelop.SaveDesktop.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+modules:
+  - name: savedesktop
+    buildsystem: simple
+    build-commands:
+       - install -Dm755 -t /app/bin run.sh
+       - install -Dm755 -t /app src/main_window.py
+       - install -D -t /app/share/applications flatpak/com.github.vikdevelop.SaveDesktop.desktop
+       - install -D -t /app/share/icons/hicolor/128x128/apps flatpak/icons/com.github.vikdevelop.SaveDesktop.png
+       - install -D -t /app/share/icons/hicolor/128x128/apps flatpak/symbolic-icons/exclamation_mark.png
+       - install -D -t /app/share/metainfo flatpak/com.github.vikdevelop.SaveDesktop.metainfo.xml
```
Comment (hfiguiere) [inline io.github.vikdevelop.SaveDesktop.yaml:26]: names don't match the app-id.

File: io.github.vikdevelop.SaveDesktop.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+runtime-version: '43'
+sdk: org.gnome.Sdk
+command: run.sh
+finish-args:
+    - --share=ipc
+    - --socket=wayland
+    - --socket=fallback-x11
+    - --device=dri
+    # for load ~/.local/share, ~/.config, ~/Downloads/SaveDesktop, .themes and .icons
+    - --filesystem=home
```
Comment (JakobDev) [inline io.github.vikdevelop.SaveDesktop.yaml:12]: Why did you not just give permission for these dirs?

**Comment by vikdevelop** [comment]
bot, build io.github.vikdevelop.SaveDesktop

**Comment by vikdevelop** [comment]
bot, build io.github.vikdevelop.SaveDesktop

**Comment by vikdevelop** [comment]
bot, build io.github.vikdevelop.SaveDesktop

**Comment by vikdevelop** [comment]
bot, build io.github.vikdevelop.SaveDesktop

**Comment by vikdevelop** [comment]
bot, build io.github.vikdevelop.SaveDesktop

**Comment by vikdevelop** [comment]
I use permissions ( `filesystem=~/.config`, `filesystem=~/.local/share`, `filesystem=~/.themes`, `filesystem=~/.icons`  ) because they are needed for load and import configuration of GNOME, Xfce, Cinnamon, Budgie and Cosmic. Permission `filesystem=xdg-download/SaveDesktop:create` is for save configuration to tar.gz archive.

**Comment by vikdevelop** [comment]
bot, build io.github.vikdevelop.SaveDesktop

**Comment by vikdevelop** [comment]
bot, build io.github.vikdevelop.SaveDesktop

**Comment by vikdevelop** [comment]
The application works without any problems on the test build, so it is ready. The reasons for using the permissions are listed in the manifest. Is there anything else I need to change?

**Comment by vikdevelop** [comment]
bot, build io.github.vikdevelop.SaveDesktop

**Comment by vikdevelop** [comment]
@hfiguere I've listed all the necessary access to the all directories which are needed for application. Is it okay?

**Pull request closed without merging**